app/questionnaire/rules/rule_evaluator.py

```diff
@@ -42,30 +42,43 @@ def __post_init__(self) -> None:
         self.operations = Operations(language=self.language)
 
     def _evaluate(self, rule: dict[str, Sequence]) -> Union[bool, Optional[date]]:
-        next_rule = next(iter(rule))
-        operator = Operator(next_rule, self.operations)
-        operands = rule[next_rule]
+        operator_name = next(iter(rule))
+        operator = Operator(operator_name, self.operations)
+        operands = rule[operator_name]
 
         if not isinstance(operands, Sequence):
             raise TypeError(
                 f"The rule is invalid, operands should be of type Sequence and not {type(operands)}"
             )
 
-        resolved_operands = self.get_resolved_operands(operands)
+        resolved_operands: Iterable[Union[bool, Optional[date], ValueSourceTypes]]
+
+        if operator_name == Operator.MAP:
+            resolved_iterables = self._resolved_operand(operands[1])
```

Comment:
`resolved_iterables ` can be `bool, Optional[date], ValueSourceTypes` , how come it is iterable ? consider renaming